p2p/lightchain.py

```diff
@@ -333,14 +333,33 @@ def callback(r):
         return rlp.decode(rlp_account, sedes=Account)
 
     @alru_cache(maxsize=1024, cache_exceptions=False)
-    async def get_contract_code(self, block_hash: Hash32, key: bytes) -> bytes:
+    async def get_contract_code(self, block_hash: Hash32, address: Address) -> bytes:
+        """
+        :param block_hash: find code as of the block with block_hash
+        :param address: which contract to look up
+
+        :return: bytecode of the contract, ``b''`` if no code is set
```

Comment:
Currently (partly because of #916), this will also return b'' if the peer we pick below doesn't have the given block. Maybe mention that here, with a fixme that we should instead raise an exception if none of our peers have that block?

Reply:
Cool, thoughts on the new approach?